lib/liquid/lexer.rb

```diff
@@ -157,81 +156,80 @@ class Lexer2
       table.freeze
     end
 
-    def initialize(input)
-      @ss = StringScanner.new(input)
-    end
-
     # rubocop:disable Metrics/BlockNesting
-    def tokenize
-      @output = []
-
-      until @ss.eos?
-        @ss.skip(WHITESPACE_OR_NOTHING)
-
-        break if @ss.eos?
-
-        start_pos = @ss.pos
-        peeked = @ss.peek_byte
-
-        if (special = SPECIAL_TABLE[peeked])
-          @ss.scan_byte
-          # Special case for ".."
-          if special == DOT && @ss.peek_byte == DOT_ORD
-            @ss.scan_byte
-            @output << DOTDOT
-          elsif special == DASH
-            # Special case for negative numbers
-            if (peeked_byte = @ss.peek_byte) && NUMBER_TABLE[peeked_byte]
-              @ss.pos -= 1
-              @output << [:number, @ss.scan(NUMBER_LITERAL)]
+    class << self
+      def tokenize(input, ss)
```

Comment:
Had a search on github, didn't see any references to `Liquid::Lexer.new` so this change should be fine